### ISSUE TITLE
Labels can be added when no classes exist

### ISSUE THREAD

**Issue by ndrezn** [mlexchange/mlex_highres_segmentation#113]
It's possible to remove all classes, and you can still annotate. When you add annotations, they appear as grey and are removed if you add a class after the fact.

I would expect either:
- It's not possible to annotate when there are no classes
- It's not possible to have no classes.
- Or some other option?

Screen recording:
https://github.com/mlexchange/mlex_highres_segmentation/assets/38958867/7c6d3598-666c-439b-8bf5-7bfa8a7f124e

**Comment by ndrezn** [comment]
cc @cleaaum

**Comment by cleaaum** [comment]
I purposely left it like this - it's the default behaviour for now because we allowed the user to delete all classes. Any drawn annotation in the grey is not saved because its not attached to a class. We weren't sure what the best option was here - but its something we can ask LBL this week. I'll implement this based on what they prefer:)

**Comment by ndrezn** [comment]
Maybe if it's drawn with no class, then when you add a class those annotations should be assigned to the first class? That way they aren't deleted.

**Comment by ndrezn** [comment]
I think the logic would be current behaviour with no classes (i.e. grey and no class defined), and when a user clicks "Add class" it would change the color of the grey annotations? But disable drawing plausibly is fine too, and certainly easier to implement.

**Comment by cleaaum** [comment]
The latter would definitely be easier to implement. For the former,  it doesn't make sense to me that these grey annotations get saved when a new class is created, its adding complexity to the code and not following the pattern previously implemented. These annotations get saved as soon as they are drawn, so this would require to redo an implementation for this edge case. When you annotate an image, you use a class, so if no class exists it makes sense that you're not annotating the image. 
If they like this approach we can implement it, but I would discourage it, and we can agree to disagree 😄 

Initially the idea was to disable the deletion of the last class, but we got rid of this in the case that the user didn't like the default class color and wanted a new color (color could not be changed by the user). In the meantime, I implemented the possibility of changing the class color, so we could add back the inability to delete the last class as well.
